src/components/GameItem.js

```diff
@@ -4,9 +4,59 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
-  propTypes = {
+  constructor(props) {
+    super(props);
+    this.state = {
+      clickDisplay: 'none',
+      clicked: false,
+    }
+  }
+ 
+  static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
+    type: PropTypes.string.isRequired,
+    updatedScoreCallback: PropTypes.func,
+  }
+
+  updateClickStatus = () => {
+    this.setState({
+      clicked: true
+    });
+  }
+
+  updateIconClass = () => {
+    this.setState({
+      clickDisplay: this.props.type === 'litter' ? 'spotted-litter' : 'spotted-nature',
+    });
+  }
+
+  onIconClick = () => {
+
+    const iconType = this.props.type;
+    const clickStatus = this.state.clicked;
+
+    let earnedPoints
+
+    // earn 1 point for litter, -1 points for nature, and zero points otherwise
+    if (clickStatus === false) {
+      if (iconType === 'litter') {
+        earnedPoints = 1;
+      } else {
+        earnedPoints = -1;
+      }
+    } else {
+      earnedPoints = 0;
+    }
+
+    // invoking callback function
+    this.props.updateScoreCallback(earnedPoints);
```

Comment:
fantaastic decision to make `earnedPoints` a parameter. This design makes it extensible. Gooood call